### PR TITLE
[c++ Frontend] Make SGD match python

### LINKED ISSUE #15530: c++ SGD and python SGD don't match
## 🐛 Bug

I would consider this a bug as the stated goals of the c++ frontend is to track the python library.

The difference is itself stated in the comments in the python library:

"
.. note::
        The implementation of SGD with Momentum/Nesterov subtly differs from
        Sutskever et. al. and implementations in some other frameworks.
        Considering the specific case of Momentum, the update can be written as
        .. math::
                  v = \rho * v + g \\
                  p = p - lr * v
        where p, g, v and :math:`\rho` denote the parameters, gradient,
        velocity, and momentum respectively.
        This is in contrast to Sutskever et. al. and
        other frameworks which employ an update of the form
        .. math::
             v = \rho * v + lr * g \\
             p = p - v
        The Nesterov version is analogously modified.
"

The C++ API implements SGD in the way that Sutskever et al and other frameworks do, which is to only apply the learning rate to the gradient, not to the entire step.

I think the discrepancy should be fixed.

cc @goldsborough

### PR DESCRIPTION
**Pull request by an-kumar**
Fixes https://github.com/pytorch/pytorch/issues/15530

### CODE REVIEW

File: torch/csrc/api/src/optim/sgd.cpp [hunk 1]
```diff
@@ -43,7 +43,7 @@ void SGD::step() {
     }
 
     NoGradGuard guard;
-    p.add_(-update);
+    p.add_(-options.learning_rate_*update);
```
Comment (goldsborough) [inline torch/csrc/api/src/optim/sgd.cpp]: nit: please add spaces around the `*`

**Review by goldsborough — APPROVED** [review]

**Comment by goldsborough** [comment]
Looks good, just one comment.

**Comment by an-kumar** [comment]
Done. not sure why tests are failing after that change though.
